plugins/test/unit/test_nectar_factory.py

```diff
@@ -9,11 +9,24 @@
 from pulp_rpm.plugins.importers.yum.repomd import nectar_factory
 
 
+class FakeLocalFileDownloader(LocalFileDownloader):
+    def __init__(self, config, event_listener):
+        pass
+
+
+class FakeHTTPTHreadedDownloader(HTTPThreadedDownloader):
+    def __init__(self, config, event_listener):
+        pass
+
+
 class NectarFactoryTests(unittest.TestCase):
     def setUp(self):
         super(NectarFactoryTests, self).setUp()
         self.mock_config = mock.MagicMock()
         self.mock_event_listener = mock.MagicMock()
+        nectar_factory.SCHEME_DOWNLOADERS['file'] = FakeLocalFileDownloader
+        nectar_factory.SCHEME_DOWNLOADERS['http'] = FakeHTTPTHreadedDownloader
+        nectar_factory.SCHEME_DOWNLOADERS['https'] = FakeHTTPTHreadedDownloader
```

Comment:
You should mock the downloader like this:
`@patch('nectar.downloaders.threaded.HTTPThreadedDownloader.__init__', Mock())`